### ISSUE TITLE
Slider with Lazy Load Images?

### ISSUE THREAD

**Issue by hansnolte** [uikit/uikit#1319]
Is it possible to create the slider with Lazy Load Images?

Do I need a Third-Party-Plugin or is it with On-Board-Plugin (maybe Scrollspy) ?

Thanks in advance
Hans

**Comment by malles** [comment]
You could try to adapt this pen for use i slider. I think it should work: http://codepen.io/malles/pen/vOXzxZ

**Comment by hansnolte** [comment]
Hi malles,

many thanks for your response.

But with my limited Javascript knowledge, I do not think I get it to run.

Do you have a tip for a external lazy loading script?

**Comment by malles** [comment]
Hmm, I'm afraid not. It might be interesting to build it in the slider component as an option. For a slider it's an obvious performance winner.
@aheinze Maybe allow a `data-source` attribute on image and lazyload those when coming into view?

**Comment by hansnolte** [comment]
"For a slider it's an obvious performance winner."
 I totally agree!

I'm just testing "lazysizes"
https://github.com/aFarkas/lazysizes

**Comment by ne-u** [comment]
+1

**Comment by Pandahisham** [comment]
+1 I think a simple plugin to lazyload any image anywhere would be an added bonus

**Comment by ne-u** [comment]
@hansnolte is lazysizes working?

**Comment by hansnolte** [comment]
Hi ne-u,
I have only tested shortly.
My customer did not want a slider anymore, but a Lightbox.

I think It's a kind of "defer".
The images are not loaded on demand, but in series.
(If I did everything right)

**Comment by aheinze** [comment]
I think this should be easily fixed by using an exiisting 3rd party lazyloader script.

**Comment by freefri** [comment]
Does anyone recommend an easy to integrate lazy load plug-in, not only for the slider but general for every image?

**Comment by viocassel** [comment]
Just use [lazysizes](https://github.com/aFarkas/lazysizes).

**Comment by ladislavsulc** [comment]
Also for lightbox videos (button on click), it loads too much JS before the video is even clicked. Any suggestions, please?

**Comment by ladislavsulc** [comment]
Okay, bend the theme in WP with Lazysizes. It works! Thank you.

**Comment by ladislavsulc** [comment]
Okay, I have discovered one problem @viocassel. When I use [lazysizes](https://github.com/aFarkas/lazysizes) it requires attribute data-src. If I use modal with an iframe (YT video) and change it to data-src, closing the modal will not stop the video. It still plays in the background. I validated it is caused by this because when I change the iframe to only **src**, it works again. I guess some JS problem connected to src and `uk-video`?

Anyone a suggestion how to solve this? I don't want to use lightbox element for this.

Thank you.
